vault/src/metrics.rs

```diff
@@ -428,48 +439,94 @@ async fn publish_issue_count<B: BitcoinCoreApi + Clone + Send + Sync>(
     }
 }
 
-async fn publish_redeem_count<B: BitcoinCoreApi + Clone + Send + Sync>(
+async fn publish_time_to_first_deadline<B: BitcoinCoreApi + Clone + Send + Sync>(
     parachain_rpc: &InterBtcParachain,
     vault_id_manager: &VaultIdManager<B>,
+    redeems: &[(H256, InterBtcRedeemRequest)],
 ) {
-    if let Ok(redeems) = parachain_rpc
-        .get_vault_redeem_requests(parachain_rpc.get_account_id().clone())
-        .await
-    {
-        for vault in vault_id_manager.get_entries().await {
-            let relevant_redeems: Vec<_> = redeems
+    for vault in vault_id_manager.get_entries().await {
+        let data: Result<_, crate::Error> = tokio::try_join!(
+            async { parachain_rpc.get_redeem_period().await.map_err(Into::into) },
+            async { parachain_rpc.get_current_chain_height().await.map_err(Into::into) },
+            async { vault.btc_rpc.get_block_count().await.map_err(Into::into) },
+        );
+        if let Ok((redeem_period, para_height, bitcoin_height)) = data {
+            let remaining_time = redeems
                 .iter()
-                .filter(|(_, redeem)| redeem.vault == vault.vault_id)
-                .map(|(_, redeem)| redeem.status.clone())
-                .collect();
-
-            vault.metrics.redeems.open_count.set(
-                relevant_redeems
-                    .iter()
-                    .filter(|status| matches!(status, RedeemRequestStatus::Pending))
-                    .count() as f64,
-            );
-            vault.metrics.redeems.completed_count.set(
-                relevant_redeems
-                    .iter()
-                    .filter(|status| matches!(status, RedeemRequestStatus::Completed))
-                    .count() as f64,
-            );
-            vault.metrics.redeems.expired_count.set(
-                relevant_redeems
-                    .iter()
-                    .filter(|status| {
-                        matches!(
-                            status,
-                            RedeemRequestStatus::Reimbursed(_) | RedeemRequestStatus::Retried
-                        )
-                    })
-                    .count() as f64,
-            );
+                .filter(|(_, redeem)| redeem.vault == vault.vault_id && redeem.status == RedeemRequestStatus::Pending)
+                .filter_map(|(_, redeem)| calculate_remaining_time(redeem_period, redeem, para_height, bitcoin_height))
+                .min();
+
+            // if no redeem deadlines, then use the redeem period
+            let remaining_time: Duration =
+                remaining_time.unwrap_or_else(|| Duration::from_millis(runtime::MILLISECS_PER_BLOCK) * redeem_period);
+
+            vault
+                .metrics
+                .remaining_time_to_redeem_hours
+                .set(remaining_time.as_secs_f64() / 3600.0);
```

Comment:
Are any of the arithmetic traits implemented on `Duration`? Might be cleaner to do that if we have `3600` as a constant.

Reply:
@sander2 can you check this? I think it may be slightly cleaner that in-code magic numbers.

Reply:
https://doc.rust-lang.org/stable/std/time/struct.Duration.html#impl-Div%3Cu32%3E